plugins/modules/aci_interface_policy_storm_control.py

```diff
@@ -0,0 +1,434 @@
+
+    storm_control_policy = module.params.get("storm_control_policy")
+    description = module.params.get("description")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+    storm_control_types = MATCH_STORM_CONTROL_POLICY_TYPE_MAPPING.get(module.params.get("storm_control_types"))
+    storm_control_action = module.params.get("storm_control_action")
+    storm_control_soak_action = module.params.get("storm_control_soak_action")
+
+    rates_input = {}
+
+    stom_control_types_configs = [
+        dict(
+            config_input=module.params.get("all_types_configuration"),
+            rates=dict(rate=dict(percentage="rate", pps="ratePps"), burst_rate=dict(percentage="burstRate", pps="burstPps")),
+        ),
+        dict(
+            config_input=module.params.get("broadcast_configuration"),
+            rates=dict(rate=dict(percentage="bcRate", pps="bcRatePps"), burst_rate=dict(percentage="bcBurstRate", pps="bcBurstPps")),
+        ),
+        dict(
+            config_input=module.params.get("multicast_configuration"),
+            rates=dict(rate=dict(percentage="mcRate", pps="mcRatePps"), burst_rate=dict(percentage="mcBurstRate", pps="mcBurstPps")),
+        ),
+        dict(
+            config_input=module.params.get("unicast_configuration"),
+            rates=dict(rate=dict(percentage="uucRate", pps="uucRatePps"), burst_rate=dict(percentage="uucBurstRate", pps="uucBurstPps")),
+        ),
+    ]
+
+    for config in stom_control_types_configs:
+        config_input = config.get("config_input")
+        rates = config.get("rates")
+        if config_input is not None:
+            if config_input.get("rate_type") == "percentage":
+                for rates_type, rates_attributes in rates.items():
+                    input = config_input.get(rates_type)
+                    if input is not None and not (0 <= float(input) <= 100):
+                        module.fail_json(
+                            msg="If argument rate_type is percentage, argument {0} needs to be a value between 0 and 100 inclusive, got {1}".format(
+                                rates_type,
+                                input,
+                            )
+                        )
+                    else:
+                        rates_input[rates_attributes.get("percentage")] = "{0:.6f}".format(float(input))
+                        rates_input[rates_attributes.get("pps")] = "unspecified"
+            elif config_input.get("rate_type") == "pps":
+                for rates_type, rates_attributes in rates.items():
+                    rates_input[rates_attributes.get("percentage")] = None
+                    rates_input[rates_attributes.get("pps")] = config_input.get(rates_type)
+        else:
+            for rates_type, rates_attributes in rates.items():
+                rates_input[rates_attributes.get("percentage")] = None
+                rates_input[rates_attributes.get("pps")] = None
```

Comment:
Why set the input to None for all items, they would have no effect correct? I would assume we can exclude from payload or are they required to be passed with None?